apps/wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -312,14 +320,52 @@ export function DappTxApprovalPage() {
         }
     }, [deserializeTxnFailed, loading, txRequest]);
 
-    const valuesContent = useMemo(() => {
+    const address = useAppSelector(({ account }) => account.address);
+    const txGasEstimationResult = useQuery({
+        queryKey: ['tx-request', 'gas-estimate', txRequest?.id, address],
+        queryFn: () => {
+            if (txRequest?.tx && address) {
+                const signer = thunkExtras.api.getSignerInstance(
+                    thunkExtras.keypairVault.getKeyPair()
+                );
+                let txToEstimate: Parameters<
+                    typeof signer.dryRunTransaction
+                >['0'];
+                const txType = txRequest.tx.type;
+                if (txType === 'v2' || txType === 'serialized-move-call') {
+                    txToEstimate = txRequest.tx.data;
+                } else {
+                    txToEstimate = {
+                        kind: 'moveCall',
+                        data: txRequest.tx.data,
+                    };
+                }
+                return signer.getGasCostEstimation(txToEstimate);
+            }
+            return null;
+        },
+    });
+    const gasEstimation = txGasEstimationResult.isSuccess
```

Comment:
I don't really think you _need_ the `isSuccess` check, the presence of data implies it.